### ISSUE TITLE
Auto-concede bug: timed out while on f11

### ISSUE THREAD

**Issue by Zzooouhh** [magefree/mage#4534]
While waiting for another player to respond, somehow I managed to time out during an opponent's main phase, despite having F11 pressed. The stalling player continued playing after I got auto-conceded. This was in 4-player FFA commander and the 4-turn cycle did roughly last 300 seconds - could it be that currently auto-concede doesn't distinguish players that are passing priority at all?

**Comment by spjspj** [comment]
@Zzooouhh This is an ongoing issue.  There is a singleton instance of response.

From #4476 
```
The manifestation of the problem was the infinite loop in waitResponseOpen which locked up the game for players due to the fact that a response (of which there is currently a singleton instance) is received from Player2 conceding whilst xmage is waiting for a response from Player1. This causes the response object to break out of its wait function in object.java.

(object.java):
    public final void wait() throws InterruptedException {
        wait(0);
    }

This wait function is always called from a response.wait(); called in waitResponseOpen in HumanPlayer.java.

The ultimate root of the problem is/was the flag responseOpenedForAnswer. It could be set to false by a concede which would then set responseOpenedForAnswer = false; (on line 222 in function prepareForResponse). This is called from when waitResponseOpen returns, breaking out of the wait procedure (even though it started waiting on Player1's response to a replacement trigger choice dialog, it then handles Player2's concede event).

I've not seen or been able to produce other situations where this problem arises apart from where someone concedes. Killing the window is handled ok (as an abort is sent).
```

**Comment by JayDi85** [comment]
Freezes on concede was fixed.

There is a `cheat` bug with choice dialogs -- on your choice timer switch to your opponent, but you control and can wait to "time out" and win (see #4349). Your opponent can be one of that cheat players (look at game history for cards with choice dialog in abilities).

**Comment by Zzooouhh** [comment]
In this scenario, neither player conceded prior to this happening, if that makes any difference. I'm leaning towards the choice dialog bug being the cause, although I'm not sure if I saved the logs (though IIRC it happened while opponent was choosing blockers for a 3rd attacking player - I'm sure I wasn't the one ticking down on the timer)

**Comment by Zerrisx** [comment]
Fixed / duplicate of #4349 depending on the cause.
